### PR TITLE
Refine timestamp calculation in AudioResamplerImpl

### PR DESCRIPTION
**Pull request by jasonzhangxx**
b/275647002

### CODE REVIEW

File: starboard/shared/starboard/player/filter/testing/audio_resampler_test.cc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+TEST_P(AudioResamplerTest, SunnyDay) {
+  scoped_ptr<AudioResampler> resampler = AudioResampler::Create(
+      source_sample_type_, source_storage_type_, source_sample_rate_,
+      destination_sample_type_, destination_storage_type_,
+      destination_sample_rate_, channels_);
+
+  int total_input_frames = 0;
+  std::vector<scoped_refptr<DecodedAudio>> outputs;
+  for (int i = 0; i < inputs_.size(); i++) {
```
Comment (kaidokert) [inline starboard/shared/starboard/player/filter/testing/audio_resampler_test.cc:128]: Tiny style nitpick: Consider a range based for loop, here and above too.

```
  for(auto input: inputs_) {
```

File: starboard/shared/starboard/player/filter/testing/adaptive_audio_decoder_test.cc [hunk 2]
```diff
@@ -244,11 +244,6 @@ class AdaptiveAudioDecoderTest
       last_decoded_audio_ = decoded_audio;
       return;
     }
-    // TODO: fix resampler timestamp issue.
-    // if (last_decoded_audio_) {
```
[the remark this answers is not in the document]
Comment (jasonzhangxx) [reply starboard/shared/starboard/player/filter/testing/adaptive_audio_decoder_test.cc:248]: The written audio inputs have timestamps from different dmp files and are not guaranteed to be  monotonic increasing. The checks are not appropriate, so I removed them.

[the remark this answers is not in the document]
Comment (xiaomings) [reply starboard/shared/starboard/player/filter/testing/adaptive_audio_decoder_test.cc:248]: In that case we should probably modify the test code to ensure that audio samples are always written with the correct timestamps (i.e. monotonic increasing) under the same playback.
Feel free to add a TODO for this so we can get this CL merged.

**Comment by xiaomings** [comment]
We should also ensure that all tests disabled due to this issue are re-enabled and pass (so prove that this actually fixes the issues on all ATV devices).  This can be a follow up CL.

**Review by xiaomings — APPROVED** [review]

**Pull request merged**